src/components/molecules/NoteDetailFolderNavigator.tsx

```diff
@@ -106,6 +115,7 @@ const NoteDetailFolderNavigator = ({
         <React.Fragment key={folderData.pathname}>
           <Icon path={mdiSlashForward} />
           <FolderNavItem
+            title={getFolderTooltip(folderData.name)}
```

Comment:
Folder tooltip should be implemented in FolderNavItem. Not here

Reply:
do you mean separating the logic into a separate component, something like this?

```ts
const FolderNavItemButton = styled.button`
  ...
`

interface FolderNavItemProps {
  text: string
  href?: string
  className?: string
  onClick?: function
}

const FolderNavItem = ({ href, text, ...props }: FolderNavItemProps) => {
  const { t } = useTranslation()
  
  const getStorageTooltip = (storageName: string) =>
    `${t('storage.storage')} ${storageName}: ${t('general.allnote')}`
  const getFolderTooltip = (foldername: string) =>
    `${t('folder.folder')} ${foldername}: ${t('general.allnote')}`

  const isFolder = typeof href !== 'string' || href.includes(`/notes/${text}/`)
  const tooltip = isFolder ? getFolderTooltip(text) : getStorageTooltip(text)

  return (
    <FolderNavItemButton title={tooltip} {...props}>
      {text}
    </FolderNavItemButton>
  )
}

<FolderNavItem
  text={folderData.name}
  onClick={() => {
    push(
      `/app/storages/${storageId}/notes${folderData.pathname}/${noteId}`
    )
  }}
  className={
    currentFolderPathname === folderData.pathname ? 'active' : ''
  }
/>
```

Reply:
The component should accept a folder, not `text` or `heref`. It should be able to resolve children and all other props for the anchor element by itself.

Reply:
Refactored in 9416a0a:

```tsx
  return (
    <Container>
      <IconContainer>
        <Icon path={mdiBookOpen} />
      </IconContainer>
      <FolderNavItem path={`/app/storages/${storageId}/notes/${noteId}`} />

      {folderDataList.map((folderData) => (
        <React.Fragment key={folderData.pathname}>
          <Icon path={mdiSlashForward} />
          <FolderNavItem
            path={`/app/storages/${storageId}/notes${folderData.pathname}/${noteId}`}
          />
        </React.Fragment>
      ))}
    </Container>
  )
```

Reply:
I'll deal with it myself. :+1: